ecl/hql/hqlutil.cpp

```diff
@@ -5807,6 +5807,10 @@ IHqlExpression * extractCppBodyAttrs(unsigned lenBuffer, const char * buffer)
                     i = end;
                     if(matchOption(start, end, buffer, 4, "pure", false, valueStart, valueEnd))
                         attrs.setown(createComma(attrs.getClear(), createAttribute(pureAtom)));
+                    else if (matchOption(start, lenBuffer, buffer, 8, "volatile", false, valueStart, valueEnd))
```

Comment:
Although I cannot see it causing problems in this case, shouldn't it use 'end' rather than 'lenBuffer'  for consistency?